### PR TITLE
[angular] Remove dependency on @types/jquery

### LINKED ISSUE #17284: @types/angular are incompatible with the latest version of @types/jquery
- [x] I tried using the `@types/angular` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
 
Please specify old version for jquery typings in dependencies, I think it should be 2.0.47

### PR DESCRIPTION
**Pull request by leonard-thieu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://docs.angularjs.org/api/ng/function/angular.element
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17284
- ~~Increase the version number in the header if appropriate.~~
- ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~~

AngularJS doesn't have a hard dependency on jQuery. It has a built-in subset of jQuery (jqLite) and will use that unless jQuery is present.

This PR removes `@types/angular`'s dependency on `@types/jquery` to match the actual implementation and fix the errors caused from breaking changes introduced in the recently published `@types/jquery` v3.

This also removes `@types/angular-animate`'s dependency on `@types/jquery`.

**Summary of changes**
- Fixes #17284
- `@types/angular`
  - Replaced `@types/jquery` dependency with jqLite
    - jqLite's declarations are copied from `@types/jquery` v2 (pared down to AngularJS's API)
  - Update tests to use only jqLite's API
- `@types/jquery`
  - Change legacy type `JQueryEventObject` to extend from legacy types instead of `JQuery.Event`
    - `JQueryEventObject` is not compatible with `JQuery.Event`
- `@types/jquery-mousewheel`
  `@types/jquery.pjax`
  `@types/royalslider`
  `@types/sharepoint`
  - Updated to target `@types/jquery` v3

### CODE REVIEW

**Comment by leonard-thieu** [comment]
@andy-ms I want to add tests to ensure `@types/angular` handles merging with `@types/jquery`. Including the test files in the project will include jQuery and that causes errors. I tried not declaring the test files in `tsconfig.json` but then `types-publisher` reports unused files as a result. What's the proper way to setup tests in this scenario?

See https://github.com/leonard-thieu/DefinitelyTyped/tree/ad96e2c5b9bc08873a6dd6c0635b9fed3e53c2f1/types/angular/test/jquery-v3-merging.

**Comment by andy-ms** [comment]
We don't support having multiple test projects, there can only be one `tsconfig.json`. You say: "Including the test files in the project will include jQuery and that causes errors", but aren't you testing that that *doesn't* cause errors?

File: types/jquery.pjax/tsconfig.json [hunk 1]
```diff
@@ -12,6 +12,11 @@
         "typeRoots": [
             "../"
         ],
+        "paths": {
```
Comment (andy-ms) [inline types/jquery.pjax/tsconfig.json]: This should only be done if the package really relies on an older version of jquery.

**Comment by leonard-thieu** [comment]
> You say: "Including the test files in the project will include jQuery and that causes errors", but aren't you testing that that doesn't cause errors?

Sorry, I should've clarified. The errors are caused because `@types/jquery` requires the `es6` lib and TypeScript 2.3 but `@types/angular` doesn't. While I _could_ add those requirements to `@types/angular`, it didn't seem appropriate to do that just to satisfy tests. I was also worried that referencing jQuery in the main project would cause it to be declared as a dependency.

**Comment by andy-ms** [comment]
One option would be to just move the tests to `jquery`. It would also be fine to just update angular to have a `// TypeScript Version: 2.3` header and use `--lib es6`.

**Comment by leonard-thieu** [comment]
Not sure why it's not building in this repo. Works in mine https://github.com/leonard-thieu/DefinitelyTyped/pull/3.

**Review by lijunle — APPROVED** [review]

**Comment by cletusw** [comment]
This should probably have been a major revision. It broke our build because we were transitively depending on the nested `@types/jquery` that we now have to go install ourselves.

**Comment by leonard-thieu** [comment]
@cletusw `@types` packages do not use semver and are instead pinned to the MAJOR.MINOR version of the package they declare types for.

**Comment by cletusw** [comment]
Oh, okay. So we should not be using `^` in our package.json then, right? Since patch versions can contain breaking changes?

**Comment by johnnyreilly** [comment]
Yup: https://blog.johnnyreilly.com/2017/02/typescript-types-and-repeatable-builds.html

**Comment by jbroni** [comment]
We're using jquery together with our angular application and we thus require both `@types/angular` and `@types/jquery`.
This is now causing our build to fail with duplicate identifiers stemming from `jqlite.d.ts`.

Do you have any suggestions to how we can use `@types/jquery` alongside `@types/angular` following this change?

**Comment by leonard-thieu** [comment]
@jbroni fixed by #17510

**Pull request merged**